src/openloco/windows/ObjectSelectionWindow.cpp

```diff
@@ -1,17 +1,1412 @@
+
+                    image = (1 << 30) | (1 << 24) | (1 << 23) | (1 << 20) | (1 << 19) | image_ids::tab;
+                    gfx::draw_image(dpi, xPos, yPos, image);
+
+                    if (row < 1)
+                    {
+                        auto colour = colour::get_shade(self->colours[1], 7);
+                        gfx::draw_rect(dpi, xPos, yPos + 26, 31, 1, colour);
+                    }
+                }
+                xPos += 31;
+            }
+        }
+    }
+
+    // 0x004B7733
+    static void drawVehicle(gfx::drawpixelinfo_t* dpi, vehicle_object* vehicleObject, uint8_t eax, uint8_t esi, gfx::point_t offset)
     {
         registers regs;
-        call(0x00472A20, regs);
+        regs.cx = offset.x;
+        regs.dx = offset.y;
+        regs.eax = eax;
+        regs.esi = esi;
+        regs.bl = colour::saturated_green;
+        regs.bh = 2;
+        regs.ebp = (uintptr_t)vehicleObject;
+        regs.edi = (uintptr_t)dpi;
+        call(0x4B7733, regs);
+    }
+
+    static const xy32 objectPreviewOffset = { 56, 56 };
+    static const gfx::ui_size_t objectPreviewSize = { 112, 112 };
+    static const uint8_t descriptionRowHeight = 10;
+
+    // 0x00473579
+    static void drawImage(objectmgr::header* header, gfx::drawpixelinfo_t* dpi, int16_t x, int16_t y, void* objectPtr)
+    {
+        auto type = header->get_type();
+        switch (type)
+        {
+            case object_type::sound:
+            case object_type::steam:
+            case object_type::town_names:
+            case object_type::cargo:
+            case object_type::climate:
+            case object_type::scenario_text:
+                // null
+                break;
+
+            case object_type::interface_skin:
+            {
+                auto object = reinterpret_cast<interface_skin_object*>(objectPtr);
+                auto image = gfx::recolour(object->img, colour::saturated_green);
+
+                gfx::draw_image(dpi, x - 32, y - 32, image);
+                break;
+            }
+
+            case object_type::currency:
+            {
+                auto object = reinterpret_cast<currency_object*>(objectPtr);
+                auto currencyIndex = (object->object_icon + 3) << 4;
+                std::printf("%d\n", currencyIndex);
+                auto defaultElement = gfx::get_g1element(0x77F0);
+                auto backupElement = *defaultElement;
+                auto currencyElement = gfx::get_g1element(currencyIndex);
+
+                gfx::get_g1element(0x77F0)->offset = currencyElement->offset;
+                gfx::get_g1element(0x77F0)->width = currencyElement->width;
+                gfx::get_g1element(0x77F0)->height = currencyElement->height;
+                gfx::get_g1element(0x77F0)->x_offset = currencyElement->x_offset;
+                gfx::get_g1element(0x77F0)->y_offset = currencyElement->y_offset;
+                gfx::get_g1element(0x77F0)->flags = currencyElement->flags;
+
```

Comment:
Lacks setting of the currency character width to the width of the image.`unk_112CBA7` should be part of an array starting at `0x112C884`